modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/enricher/StudioModelEnricher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection.enricher;
+
+import org.mule.extension.annotation.api.capability.Studio;
+import org.mule.extension.api.introspection.declaration.DescribingContext;
+import org.mule.extension.api.introspection.declaration.fluent.DeclarationDescriptor;
+import org.mule.extension.api.introspection.property.StudioModelProperty;
+import org.mule.module.extension.internal.introspection.ImmutableStudioModelProperty;
+import org.mule.module.extension.internal.model.AbstractAnnotatedModelEnricher;
+
+/**
+ * Enriches the {@link DeclarationDescriptor} with a model property which key is {@value StudioModelProperty#KEY} and the value an instance of {@link StudioModelProperty}.
+ * <p/>
+ * If the extension is annotated with {@link Studio} and it is not marked as provided, the values of it will be used to initialize the {@link StudioModelProperty}. If it is not annotated with it, the model will be enriched with the default values
+ *
+ * @since 4.0
+ */
+public final class StudioModelEnricher extends AbstractAnnotatedModelEnricher
```

Comment:
this class' logic should be modified per the comments on the isProvided() semantics